presto-main/src/main/java/com/facebook/presto/operator/SimpleMergeSortComparator.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.SortOrder;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class SimpleMergeSortComparator
+        implements MergeSortComparator
+{
+    public static class Factory
+            implements MergeSortComparatorFactory
+    {
+        @Override
+        public MergeSortComparator create(List<Type> sortTypes, List<Integer> sortChannels, List<SortOrder> sortOrders)
+        {
+            return new SimpleMergeSortComparator(sortTypes, sortChannels, sortOrders);
+        }
+    }
+
+    private final List<Integer> sortChannels;
+    private final List<SortOrder> sortOrders;
+    private final List<Type> sortTypes;
+
+    public SimpleMergeSortComparator(List<Type> sortTypes, List<Integer> sortChannels, List<SortOrder> sortOrders)
```

Comment:
I don't like that we pass here sortTypes. I would rather pass here all types, and then get the actual type by the sort channel.